core/modules/qdisp/QueryResource.cc

```diff
@@ -44,46 +44,53 @@
 #include "qdisp/JobStatus.h"
 #include "qdisp/QueryRequest.h"
 #include "qdisp/QueryResource.h"
-#include "qdisp/ResponseRequester.h"
 
 namespace lsst {
 namespace qserv {
 namespace qdisp {
 
+QueryResource::QueryResource(std::shared_ptr<JobQuery> jobQuery)
+      // this char* must live as long as this object, so copy it on heap
+    : Resource(::strdup(jobQuery->getDescription().resource().path().c_str())),
+      _xrdSsiSession(NULL), _jobQuery(jobQuery) {
+    if (rName == NULL) {
+        throw std::bad_alloc();
+    }
+}
+
+QueryResource::~QueryResource() {
+    LOGF_DEBUG("~QueryResource()");
+    std::free(const_cast<char *>(rName)); // clean up heap allocated resource path copy
+}
+
 /// May not throw exceptions because the calling code comes from
 /// xrootd land and will not catch any exceptions.
 void QueryResource::ProvisionDone(XrdSsiSession* s) { // Step 3
-    LOGF_INFO("Provision done");
+    _provisionDoneHelper(s);
```

Comment:
Fair enough.  RAII pattern would also work.  Personal taste -- I'm okay with it either way -- maybe add a quicky comment about why helper -- when I saw it I was expecting it was being called from more than one place?